mobile/library/common/http/client.cc

```diff
@@ -296,11 +302,11 @@ void Client::DirectStreamCallbacks::onError() {
   // TODO(goaway): What is the expected behavior when an error is received, held, and then another
   // error occurs (e.g., timeout)?
 
-  error_ = streamError();
   if (explicit_flow_control_ && response_headers_forwarded_ && bytes_to_send_ == 0) {
     ENVOY_LOG(debug, "[S{}] defering remote reset stream due to explicit flow control",
               direct_stream_.stream_handle_);
-    if (!remote_end_stream_received_) {
+    if (direct_stream_.parent_.getStream(direct_stream_.stream_handle_,
```

Comment:
this is a better check for "is the stream closed" since it handles the case where we get reset twice.